torch/nn/attention/flex_attention.py

```diff
@@ -713,13 +713,33 @@ def _create_empty_block_mask(query: Tensor, key: Tensor) -> BlockMask:
     )
 
 
+def _create_and_apply_kernel_params(query, key, value, **kwargs: Dict[str, Any]):
+    kernel_params: Dict[str, Any] = {}
+    kernel_params.update(kwargs)
+    if "scale" not in kernel_params:
+        kernel_params["scale"] = 1.0 / math.sqrt(query.size(-1))
+    if "rows_guaranteed_safe" not in kernel_params:
+        kernel_params["rows_guaranteed_safe"] = False
+    if "prescale_qk" not in kernel_params:
+        kernel_params["prescale_qk"] = False
+
+    # If foward kernel needs to return logsumexp is decided by this rule internally.
+    any_inputs_require_grad = (
+        query.requires_grad or key.requires_grad or value.requires_grad
+    )
+    output_logsumexp = any_inputs_require_grad and torch.is_grad_enabled()
+    kernel_params["output_logsumexp"] = output_logsumexp
+
+    return kernel_params
+
+
 def flex_attention(
     query: Tensor,
     key: Tensor,
     value: Tensor,
     score_mod: Optional[_score_mod_signature] = None,
     block_mask: Optional[BlockMask] = None,
-    scale: Optional[float] = None,
+    **kwargs: Dict[str, Any],
```

Comment:
I would prefer, `(args, * scale, rows_safe,..) if we want them to be kwarg only and then we actually specify them. It is much easier for discovery
I am also okay with having a 
`
*
scale = ..
kernel_options = Dict{}
`
where we just describe the options so that we can stay generic for future args. These will likely be much less frequently interacted with than scale

Reply:
I agree that `scale` should be separate.

Reply:
@drisspg For your option 1, I think we need to change all dispatched mode's signature every time if we are going to add a new kernel option. So I'll go with option 2, which I think it's more friendly for extension.